__tests__/hooks/useEtherspotHistory.test.js

```diff
@@ -71,4 +63,54 @@ describe('useEtherspotHistory()', () => {
     const transactionPolygon = await result.current.getAccountTransaction('0x42');
     expect(transactionPolygon).toEqual(undefined);
   });
-})
+
+  it('returns transaction status for a given hash', async () => {
+    const transactionStatus = {
+      connextscanUrl: 'https://connextscan.io/tx/0x123',
+      status: 'completed',
+      transactionHash: '0x123',
+      transferId: 'abc123',
+    };
+
+    const wrapper = ({ children }) => <EtherspotTransactionKit provider={provider}>{children}</EtherspotTransactionKit>;
+
+    const { result } = renderHook(({ chainId }) => useEtherspotHistory(chainId), {
+      initialProps: { chainId: 1 },
+      wrapper,
+    });
+
+    // wait for history to be fetched for chain ID 1
+    await waitFor(() => expect(result.current).not.toBeNull());
+
+    // all correct props
+    const transaction1 = await result.current
+      .getAccountTransactionStatus(1, 137, '0x123')
+      .catch((e) => {
+        console.error(e);
+        return e;
+      });
+
+    expect(transaction1).toEqual(transactionStatus);
+
+    // transaction not found or does not exist
+    const transaction2 = await result.current
+      .getAccountTransactionStatus(4, 137, '0x123')
+      .catch((e) => {
+        console.error(e);
+        return e;
+      });
+
+    expect(transaction2).toEqual({})
+
+    // missing props
+    const transaction3 = await result.current
+      .getAccountTransactionStatus(4, 137)
+      .catch((e) => {
+        console.error(e);
+        return e;
+      });
+
+    expect(transaction3).toBeNull()
```

Comment:
If we're missing required properties here, should we not return an error to let the developer know they have missed something (as opposed to `null`)?

Reply:
Yes I've changed that now, I kept it simple